tests/runtime-trace-tests/tracerunner.ts

```diff
@@ -0,0 +1,309 @@
+import * as path from 'path';
+import { exec } from 'child_process';
+
+import "mocha";
+import * as chai from "chai";
+
+import { TestHost } from "../common/testHost";
+import * as util from "../common/testUtils";
+import { resolve } from 'url';
+import { promisify } from 'util';
+
+// TODO: Split this file up. I (dazuniga) spent ~1hr trying to split this file into two pieces but I cannot figure out 
+// how to make that work with the way we're using jake & namespaces / modules
+
+// setup
+function initGlobals() {
+    Promise = require("bluebird");
+    let g = global as any
+    g.pxt = pxt;
+    g.ts = ts;
+    g.pxtc = pxtc;
+    g.btoa = (str: string) => new Buffer(str, "binary").toString("base64");
+    g.atob = (str: string) => new Buffer(str, "base64").toString("binary");
+}
+initGlobals();
+
+// Just needs to exist
+pxt.setAppTarget(util.testAppTarget);
+
+// tests
+const casesDir = path.join(process.cwd(), "tests", "runtime-trace-tests", "cases");
+
+describe("convert between ts<->py ", () => {
+    // TODO: can this be moved to a mocha before() block?
+    let tsAndPyFiles: string[]
+
+    pxsim.initCurrentRuntime = pxsim.initBareRuntime
+    cleanup()
+
+    const isDisabled = (file: string): boolean =>
+        path.basename(file).indexOf("TODO") >= 0
+
+    let tsFiles = util.getFilesByExt(casesDir, ".ts")
+    let pyFiles = util.getFilesByExt(casesDir, ".py")
+    tsAndPyFiles = tsFiles.concat(pyFiles)
+        .filter(f => !isDisabled(f))
+
+    tsAndPyFiles.forEach(file => {
+        it("should preserve the runtime semantics of " + path.basename(file), async function () {
+            this.timeout(10000)
+            await testTsOrPy(file)
+            return
+        });
+    });
+});
+
+async function testTsOrPy(tsOrPyFile: string): Promise<void> {
+    let ext = path.extname(tsOrPyFile)
+    let isPy = ext === ".py"
+    let isTs = ext === ".ts"
+    if (!isPy && !isTs)
+        return Promise.reject(new Error("Invald non-.py, non-.ts file: " + tsOrPyFile))
+
+    let baselineFile: string;
+    let baseline: string;
+    let recordBaseline = (bl: string) => {
+        baseline = bl
+        baselineFile = tsOrPyFile + ".baseline"
+        fs.writeFileSync(baselineFile, baseline)
+    }
+    if (isPy) {
+        let pyFile = tsOrPyFile
+        recordBaseline(await PY(pyFile))
+        // TODO(dazuniga): py2ts doesn't work in unit tests yet, once it does enable it here
+        // let tsFile = await testPy2Ts(pyFile)
+        // await testSts(tsFile)
+        // let pyFile2 = await testTs2Py(tsFile)
+    } else {
+        let tsFile = tsOrPyFile
+        recordBaseline(await TS(tsFile))
+        await testSts(tsFile)
+        let pyFile = await testTs2Py(tsFile)
+        // TODO(dazuniga): py2ts doesn't work in unit tests yet, once it does enable it here
+        // let tsfile2 = await testPy2Ts(pyFile)
+        // await testSts(tsfile2)
+    }
+
+    return;
+    // DSL for the tests
+    type Trace = string
```

Comment:
If you want the compiler to show typing errors for Trace and File you can do it like [this](https://www.typescriptlang.org/play/#src=interface%20Trace%20extends%20String%20%7B%0D%0A%20%20%20%20_traceBrand%3F%3A%20any%3B%0D%0A%7D%0D%0A%0D%0Ainterface%20File%20extends%20String%20%7B%0D%0A%20%20%20%20_fileBrand%3F%3A%20any%3B%0D%0A%7D%0D%0A%0D%0Afunction%20foo()%3A%20Trace%20%7B%0D%0A%20%20%20%20return%20%22%22%3B%0D%0A%7D%0D%0A%0D%0Alet%20f%3A%20File%20%3D%20foo()%3B)

Reply:
Oh cool! And the "_traceBrand?: any;" is needed to avoid structural typing from seeing File and Trace as interchangeable I'm guessing ?

Reply:
Yep! This is a trick they use a lot in the TypeScript compiler